airbyte-integrations/connectors/source-stripe/source_stripe/error_handlers/stripe_error_handler.py

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Optional, Union
+
+import requests
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.error_handlers import HttpStatusErrorHandler
+from airbyte_cdk.sources.streams.http.error_handlers.response_models import ErrorResolution, ResponseAction
+
+STRIPE_ERROR_CODES = {
+    "more_permissions_required": "This is most likely due to insufficient permissions on the credentials in use. "
+    "Try to grant required permissions/scopes or re-authenticate",
+    "account_invalid": "The card, or account the card is connected to, is invalid. You need to contact your card issuer "
+    "to check that the card is working correctly.",
+    "oauth_not_supported": "Please use a different authentication method.",
+}
+
+DOCS_URL = f"https://docs.airbyte.com/integrations/sources/stripe"
+DOCUMENTATION_MESSAGE = f"Please visit {DOCS_URL} to learn more. "
+
+
+class StripeErrorHandler(HttpStatusErrorHandler):
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]] = None) -> ErrorResolution:
+        status_code = response_or_exception.status_code
+        if status_code in (requests.codes.bad_request, requests.codes.forbidden):
+            parsed_error = response_or_exception.json()
+            error_code = parsed_error.get("error", {}).get("code")
+            error_message = STRIPE_ERROR_CODES.get(error_code, parsed_error.get("error", {}).get("message"))
+            if error_message:
+                reason = f"The endpoint {response_or_exception.url} returned {status_code}: {response_or_exception.reason}. {error_message}.  {DOCUMENTATION_MESSAGE} "
+                response_error_message = HttpStream.parse_response_error_message(response_or_exception)
+                if response_error_message:
+                    reason += response_error_message
+
+                return ErrorResolution(response_action=ResponseAction.IGNORE, failure_type=FailureType.config_error, error_message=reason)
+        return super().interpret_response(response_or_exception)
```

Comment:
I think it could be interesting to document that Stripe will now retry on 5XX issues for example which is nice. Stripe API is pretty robust in that sense so I'm don't think it's improve reliability by that much but it still feels important

Reply:
I believe this was already the behavior when running some testing on the master branch unless I'm missing something. Also, I suspect this because some tests are set to "receive 500, then retry and get 200" routine.

![image](https://github.com/user-attachments/assets/15a459ee-b1b1-4c50-9d83-81bf55c2428e)

![image](https://github.com/user-attachments/assets/7158168a-48f3-4195-aabc-81e39ca32f60)

![image](https://github.com/user-attachments/assets/699c05c1-e40e-45c5-a173-e6d696cb297d)

Reply:
You are right, this is my mistake (see https://github.com/airbytehq/airbyte/blob/d0fd57ebf83faa52875b16bc67da8cccc73cbe2f/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L254-L264). So the added retries should be on only those that are not 429 and 5XX in https://github.com/airbytehq/airbyte/blob/4bcf36c80ad0dcf505923e84f7f3664c8523bac1/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/default_error_mapping.py#L11 Seems like a smaller scope so I'm fine with not documenting